test/ActivityRepository-test.js

```diff
@@ -57,21 +57,33 @@ describe('UserRepository', function() {
     expect(activityRepo).to.be.an.instanceOf(ActivityRepository);
   });
 
-  it('should store activity logs for all users', function() {
-    expect(activityRepo.userLogs[0]).to.be.an.instanceOf(ActivityLog);
-    expect(activityRepo.userLogs[0].userID).to.equal(1);
-    expect(activityRepo.userLogs[1]).to.be.an.instanceOf(ActivityLog);
-    expect(activityRepo.userLogs[1].userID).to.equal(2);
-    expect(activityRepo.userLogs[2]).to.be.an.instanceOf(ActivityLog);
-    expect(activityRepo.userLogs[2].userID).to.equal(3);
+  it('should organize user activity by date', function() {
```

Comment:
So, unlike in your version of the `SleepRepository` expected in your tests, these tests expect activity data to be processed into the `allUserLog` property as an object with dates as keys, with values that are objects with users as keys, with values that are in turn objects with each stat category as keys. So basically:
```
{
   YYYY/MM/DD: {
       1: {steps: x, minutes: x, stairs: x},
       2: {steps: x, minutes: x, stairs: x}
   },
   YYYY/MM/DD: {
       1: {steps: x, minutes: x, stairs: x},
       2: {steps: x, minutes: x, stairs: x}
   }
}
```

Reply:
This was a kind of intuitive hierarchy for me given the nature of the data, and because the methods for this class are all calculations for all users on a single date...but then again it might not help DRY out the logic of the methods at all. If that's the case, you can always trade the bulk of these tests for one test that expects the `allUserLog` to `deep.equal(activityData)` (as you did for `SleepRepo`)

Reply:
This makes sense but I think that it's unnecessary. I think accessing those values could be accomplished with a cleaner/more readable notation in a method. Such as 
```const someName = this.allUsersActivity.reduce((acc, entry) => { 
          if (entry.date === someDate) {
            acc += entry.flightsOfStairs
          } 
          return acc
      }, 0);
      return someName / this.allUsersActivity.length;
```

this may not actually be the right block to get it done, but just as an example, I think this is a cleaner and more readable way to sort the files within the method

Reply:
Yes, this is about what I've been to doing to pass your `SleepRepository` tests. I'll simplify these tests and push.